### DISCUSSION TITLE
Add CodeSee to airflow repo

### DISCUSSION THREAD

**Discussion by xinbinhuang** [apache/airflow#33591] [Ideas]
### Description

[CodeSee](https://docs.codesee.io/en/latest/) seems to be a good tool to help collaboration and onboard new contributors. It generates a map to help understand and navigate the codebase. Some use cases I have in mind:
- Identify hot/cold spots to understand what area the community is currently working on or what areas need more love
- An architecture diagram to help onboard new contributors 

Curious on how other people think about this?

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Yeah it's a discussion not issue @xinbinhuang :) so @eladkal rightfully converted it.

I think if someone could get a sample of how it could look like, it might be interestign, but I have an immediate "No" if tooling like that is only accessible via "request a demo" link or need to authorize it to do something I do not want.

The only way we could use it is if it is available to us for absolutely free, without string attached and without having to have meetings and demos scheduled. 

If they had a free OSS version when one could connect our project without any authorization, without providing personally identifiable data etc. so that we can see what it brings before we committ to i, that would be interesting thing to try.

When I try to connect to it with Github and see: 

```
Verify your GitHub identity (potiuk)
Know which resources you can access
Act on your behalf.  # <---- NO. Abslutely NO.
```

This is immediate NO. It's also against the ASF rules - all the repositories that are ASF ones need to have all the apps authorized by the infrastructure, and I am afrais "Act on your behalf" for committers is a big and red NO from the infra team.

**Reply by potiuk** [reply]
And just to add a bit. Everything we do is public. You need exactly 0 permissions to read everything that happens in our repo. I find it pretty mindboggling why just to "see" things (`CodeSee`) someone would expect an authorization to "act" on my behalf. It's a "view" on what's going on, so there should be absolutely no need to "act".
